rpc/src/rpc_subscriptions.rs

```diff
@@ -785,6 +874,81 @@ impl RpcSubscriptions {
         }
     }
 
+    fn notify_block_update(
```

Comment:
I haven't looked at this closely yet, as it likely will fold into notify_accounts_logs_programs_signatures if we merge block subscriptions into the commitment/gossip watchers lists.

Reply:
That said, I think we'll likely want some logic like this to ensure that the block tx metadata is done being written to gate notifying: https://github.com/solana-labs/solana/blob/a400b5e63dbfeed6969858472ce1c88c49defe4e/rpc/src/rpc.rs#L963

Reply:
consolidating into one call, but probably wont need that check since it uses the `get_complete_block` fn to fetch the block which I believe does a similar check by calling `if slot_meta.is_full() {`

Reply:
Ah, sorry, "meta" is a little overloaded. `slot_meta.is_full()` means that the node has received all the shreds in that slot (tracked by the SlotMeta column).
The transaction metadata -- such as transaction results, pre- and post-balances, rewards, inner instructions, and logs, etc -- gets written separately (by the TransactionStatusService) when the slot is processed, and added to the block here: https://github.com/solana-labs/solana/blob/9da826421a6f8a9046ddec533d1486abda32ffe3/ledger/src/blockstore.rs#L1916 That method doesn't error if the status/metadata isn't populated; just returns None. So `check_status_is_complete` that I linked above is used to determine that data is fully written.